File: go/problems/running_sum_of_array.go
package problems

// https://leetcode.com/problems/running-sum-of-1d-array/
func GetRunningSum(nums []int) []int {
	sums := make([]int, len(nums))
	sum := 0
	for i, num := range nums {
		sum = sum + num
		sums[i] = sum
	}
	return sums
}

File: rust/src/problems/running_sum_of_array.rs
// https://leetcode.com/problems/running-sum-of-1d-array/
pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
    let mut sum: i32 = 0;
    return nums
        .iter()
        .map(|x| {
            sum = sum + x;
            return sum;
        })
        .collect();
}